src/Transaction.java

```diff
@@ -19,6 +22,15 @@ public void setMoney(long money) {
         this.money = money;
     }
 
+    public void setDate(Date date) {
+        if (date == null) {
+            this.date = new Date();
+        }
+        else {
```

Comment:
REFACTOR: Could we not set the date in the field instead? `private Date date = new Date();` and then only have one check in this method:
```java
if (date != null) {
    this.date = date;
}
```
I might be missing something